nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -334,6 +329,14 @@ def add_electrodes(
                 index = isinstance(data[0], (list, np.ndarray))
                 elec_columns[prop].update(description=prop, data=data, index=index)
 
+    # Add the property of channel name
+    channel_name_array = checked_recording.get_channel_ids()
+    elec_columns["channel_name"].update(
+        description="a string named referencefor the channel",
+        data=channel_name_array,
+        index=False,
+    )
```

Comment:
One thing to check here would be that `channel_name` is unique within the table. The `id` field does this naturally via PyNWB using an `enforce_unique_values` type of argument which complains if you try to add an `id` that already exists. This will also keep the mapping of `channel_name` -> `id` one-to-one.

Reply:
What should we do if the strings channel_ids are not unique? 
The current implementation that I have here will always produce unique electrode.table ids but not unique channel_name. Should we throw an error?

Reply:
The `channel_ids` should be already enforced to be unique in spikeinterface

Reply:
Is this new, the following code does not throw an error?
```python
from spikeinterface.core.testing_tools import generate_recording
rec = generate_recording(num_channels=4, durations=[10])
channel_ids = rec.get_channel_ids()
non_int_channel_ids = ['a', 'b', 'c', 'a']
rec_non_int_channels = rec.channel_slice(channel_ids=channel_ids, renamed_channel_ids=non_int_channel_ids)
rec_non_int_channels.get_channel_ids()

output:
array(['a', 'b', 'c', 'a'], dtype='<U1')
```